### PR TITLE
Sampling only Remote Traces

### PR DESCRIPTION
**Pull request by renaz6**
- By default, local traces are not sampled
- Configuration added so that local tracing can be turned on if desired

### CODE REVIEW

**Comment by renaz6** [comment]
Documentation on `Parent Based Sampling` can be found here: https://github.com/xmidt-org/candlelight/pull/100

File: tracerProvider.go [hunk 1]
```diff
@@ -57,13 +57,23 @@ func ConfigureTracerProvider(config Config) (trace.TracerProvider, error) {
...
 		return nil, fmt.Errorf("%w for provider %s", ErrTracerProviderNotFound, config.Provider)
 	}
-	provider, err := providerConfig(config)
+
+	// Setting up trace sampler based on SampleLocalTraces value in config file
+	// Default behavior: never sample local traces
+	sampler := sdktrace.ParentBased(sdktrace.NeverSample())
+
+	if sampleLocalTraces {
+		sampler = sdktrace.ParentBased(sdktrace.AlwaysSample())
```
Comment (renaz6) [inline tracerProvider.go:73]: I'm not sure if we should use `AlwaysSample()` even if `sampleLocalTraces` is configured to be `true`. 
An alternative option is using `TraceIDRatioBased()` (https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#TraceIDRatioBased) and only sending a percentage through.

Comment (schmidtw) [reply tracerProvider.go:73]: If we split apart the choices a bit more we can let the user decide what they want without us needing to do much & we can expand/grow later.

Replace `sampleLocalTraces` with:
```yaml
  sampling-strategy:
    parent-initiated: ignore # [ignore (default), honor]
    no-parent: never         # [never (default), always, traceidratio, ...]
    traceidratio:            # For each sampler that needs inputs they get enumerated separately.
        percentage: 15.5
```

Choosing the quietest mode (completely off) as defaults.

Until we need to add `traceidratio` we can just know it goes here.  I think @Sachin4403 may want:
```yaml
  sampling-strategy:
    parent-initiated: honor
    no-parent: always
```

We'll probably want to use:

```yaml
  sampling-strategy:
    parent-initiated: honor
    no-parent: never
```

We can add the `traceidratio` later if someone wants/needs it.

**Review by johnabass — APPROVED** [review]
Given Wes's comment, we can do a separate PR to address the separate choices.

**Review by denopink — APPROVED** [review]

File: tracerProvider.go (hunk 1 above)
Comment (renaz6) [reply tracerProvider.go:73]: Taking care of this in #102

**Pull request merged**